blockchain/blockchain.go

```diff
@@ -59,48 +58,48 @@ func (k *BlockDbKey) UnmarshalBinary(data []byte) error {
 
 // Blockchain is responsible for keeping track of all things related to the StarkNet blockchain
 type Blockchain struct {
-	l *sync.RWMutex
-
-	head     *core.Block
 	network  utils.Network
 	database db.DB
 }
 
-func NewBlockchain(database db.DB, network utils.Network) (*Blockchain, error) {
-	chain := &Blockchain{
-		l:        &sync.RWMutex{},
+func NewBlockchain(database db.DB, network utils.Network) *Blockchain {
+	return &Blockchain{
 		database: database,
 		network:  network,
 	}
-	txn := chain.database.NewTransaction(false)
+}
+
+// Height returns the latest block height. If blockchain is empty nil is returned.
+func (b *Blockchain) Height() *uint64 {
+	headBlock, err := b.Head()
+	if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
+		panic(fmt.Sprintf("failed to retrieved head block: %v", err))
```

Comment:
was there a specific reason you didnt want to return an error instead of panicing here?